tests/code_sharing/test_ng_generate_e2e.py

```diff
@@ -0,0 +1,97 @@
+"""
+Tests for `ng g` during livesync.
+"""
+import os
+import unittest
+
+from core.base_test.tns_run_test import TnsRunTest
+from core.enums.app_type import AppType
+from core.enums.os_type import OSType
+from core.enums.platform_type import Platform
+from core.settings import Settings
+from core.utils.file_utils import Folder, File
+from products.angular.ng import NG, NS_SCHEMATICS
+from products.nativescript.tns import Tns
+from products.nativescript.tns_assert import TnsAssert
+from products.nativescript.tns_logs import TnsLogs
+from products.nativescript.tns_paths import TnsPaths
+
+
+class NGGenE2ETestsNS(TnsRunTest):
+    ns_app_name = Settings.AppName.DEFAULT
+    shared_app_name = 'SharedApp'
+    ns_app_path = TnsPaths.get_app_path(app_name=ns_app_name)
+    shared_app_path = TnsPaths.get_app_path(app_name=shared_app_name)
+
+    @classmethod
+    def setUpClass(cls):
+        TnsRunTest.setUpClass()
+        NG.kill()
+
+    def setUp(self):
+        TnsRunTest.setUpClass()
+        NG.kill()
+
+    def tearDown(self):
+        NG.kill()
+        TnsRunTest.tearDown(self)
+
+    def test_100_ns_generate_during_run_android(self):
+        NGGenE2ETestsNS.workflow(app_name=self.ns_app_name, device=self.emu, platform=Platform.ANDROID, shared=False)
+
+    @unittest.skipIf(Settings.HOST_OS != OSType.OSX, 'Skip iOS tests on non macOS machines.')
+    def test_100_ns_generate_during_run_ios(self):
+        NGGenE2ETestsNS.workflow(app_name=self.ns_app_name, device=self.sim, platform=Platform.IOS, shared=False)
+
+    def test_200_shared_generate_during_run_android(self):
+        NGGenE2ETestsNS.workflow(app_name=self.ns_app_name, device=self.emu, platform=Platform.ANDROID, shared=True)
+
+    @unittest.skipIf(Settings.HOST_OS != OSType.OSX, 'Skip iOS tests on non macOS machines.')
+    def test_200_shared_generate_during_run_ios(self):
+        NGGenE2ETestsNS.workflow(app_name=self.ns_app_name, device=self.sim, platform=Platform.IOS, shared=True)
+
+    @staticmethod
+    def workflow(app_name, device, platform, shared):
+        # Create an app
+        app_path = TnsPaths.get_app_path(app_name=app_name)
+        Folder.clean(app_path)
+        NG.new(collection=NS_SCHEMATICS, project=app_name, shared=shared)
+        TnsAssert.created(app_name=app_name, app_data=None)
+
+        # Run app initially
+        result = Tns.run_android(app_name=app_name, emulator=True, hmr=True)
```

Comment:
shouldn't this be Tns.run,  not Tns.run_android?

Reply:
Yep, just fixed by https://github.com/NativeScript/nativescript-tooling-qa/pull/60/commits/6c0831162402c60c9f5f44cad5bbe33f797dbd7c